### PR TITLE
#0: fix eager_package_silicon ND segfaults

### PR DESCRIPTION
**Pull request by aliuTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/9520
### Problem description
- CI issue on eager package silicon

### What's changed
- Fixed eager package test
- 
### Checklist
- [x] Post commit CI passes
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/device.cpp [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace ttnn {
 namespace device {
 
 Device &open_device(int device_id, size_t l1_small_size, size_t trace_region_size) {
-    tt::DevicePool::initialize({device_id}, 1, l1_small_size, trace_region_size);
+    tt::DevicePool::initialize({device_id}, 1, l1_small_size, trace_region_size, {}, true);
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/device.cpp:12]: what those `{}` and `true` are?

Comment (aliuTT) [reply ttnn/cpp/ttnn/device.cpp:12]: `l1_small_remap` and `skip_remote_device flag`.

Comment (ayerofieiev-tt) [reply ttnn/cpp/ttnn/device.cpp:12]: Do you have full control over this code? 
Can we change `skip_remote_device` from bool to enum instead?

Comment (aliuTT) [reply ttnn/cpp/ttnn/device.cpp:12]: I added the DevicePool, so I could. Any reason you want an enum over a bool?

Comment (aliuTT) [reply ttnn/cpp/ttnn/device.cpp:12]: I could also have it be inferred in the DevicePool init logic. So if device_ids is size 1 and requested device is mmio, I'll just skip by default. This is making more sense to me as I'm typing it out, so I think I'll make that change

File: tests/end_to_end_tests/conftest.py [hunk 2]
```diff
@@ -13,7 +13,10 @@
 
 @pytest.fixture(scope="function")
 def first_grayskull_device():
-    return tt_lib.device.CreateDevice(0)
+    device = tt_lib.device.CreateDevice(0)
+    yield device
+
+    tt_lib.device.CloseDevice(device)
```
Comment (tt-rkim) [inline tests/end_to_end_tests/conftest.py:19]: Lol clearly I'm a fucking BRICK

Comment (tt-rkim) [reply tests/end_to_end_tests/conftest.py:19]: No need for the `DeallocateBuffers` etc ?

Comment (aliuTT) [reply tests/end_to_end_tests/conftest.py:19]: No, in the interest of keeping things minimal I didn't pull that logic here. The end to end tests don't use Buffers

Comment (tt-rkim) [reply tests/end_to_end_tests/conftest.py:19]: what's the best way to keep an eye on whether resources get split up like buffers like this?
Just check root conftest fixture?

Comment (aliuTT) [reply tests/end_to_end_tests/conftest.py:19]: Yea.. I like the concept of keeping the testing here to a minimal. Just a refactor/test updating mistake.

**Review by tt-rkim — APPROVED** [review]

**Review by TT-BrianLiu — APPROVED** [review]

**Review by xanderchin — APPROVED** [review]

File: ttnn/cpp/ttnn/device.cpp [hunk 3]
```diff
@@ -9,7 +9,7 @@ namespace ttnn {
 namespace device {
 
 Device &open_device(int device_id, size_t l1_small_size, size_t trace_region_size) {
-    tt::DevicePool::initialize({device_id}, 1, l1_small_size, trace_region_size);
+    tt::DevicePool::initialize({device_id}, 1, l1_small_size, trace_region_size, {});
```
Comment (cfjchu) [inline ttnn/cpp/ttnn/device.cpp:12]: I think you can revert this change and constructor default value take care of it

Comment (aliuTT) [reply ttnn/cpp/ttnn/device.cpp:12]: shoot, missed this. I should be coming back to this initialize call soon. I'm planning to remove the l1_bank_remap altogether from device, and that should clean this up as well.
